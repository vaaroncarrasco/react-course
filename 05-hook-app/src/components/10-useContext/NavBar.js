import React from 'react'
import { NavLink } from 'react-router-dom';

// Link -> like a
// NavLink -> like a but with exact & activeClassName attribute

export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">

            <NavLink to="/" className="navbar-brand">useContext</NavLink>

            <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                <div className="navbar-nav">
                    <NavLink exact activeClassName="active" to="/" className="nav-item nav-link">Home</NavLink>
                    <NavLink exact activeClassName="active" to="/about" className="nav-item nav-link">About</NavLink>
                    <NavLink exact activeClassName="active" to="/login" className="nav-item nav-link">Login</NavLink>
                </div>

            </div>

        </nav>
    )
}
