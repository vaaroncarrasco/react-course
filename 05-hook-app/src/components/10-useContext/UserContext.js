import { createContext } from 'react';

// * Higher Order Component

// 1. createContext()
export const UserContext = createContext(null);