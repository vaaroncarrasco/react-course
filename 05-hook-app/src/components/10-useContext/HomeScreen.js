import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const HomeScreen = () => {

    // 3. useContext -> use createdContext to get data -> import context component and useContext()
    const { user } = useContext(UserContext);

    return (
        <div>
            <h1>HomeScreen</h1>
            <hr />

            <pre>
                {
                    JSON.stringify( user, null, 3)
                }
            </pre>
        </div>
    )
}
