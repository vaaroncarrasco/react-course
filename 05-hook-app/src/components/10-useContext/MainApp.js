import React, { useState } from 'react'
import { AppRouter } from './AppRouter'
import { UserContext } from './UserContext'

export const MainApp = () => {

    const [user, setUser] = useState({});

    return (
        // * Higher Order Component creates -> Context.Provider -> allowing to pass data thru child components
        // 2. Send data thru context component.Provider value={ value }
        <UserContext.Provider value={{
            user,
            setUser
        }}>

            <AppRouter />

        </UserContext.Provider>
    )
}
