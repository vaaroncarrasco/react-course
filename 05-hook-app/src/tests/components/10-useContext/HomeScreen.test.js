import React from 'react';
import { mount } from 'enzyme';
import { HomeScreen } from '../../../components/10-useContext/HomeScreen';
import { UserContext } from '../../../components/10-useContext/UserContext';

describe('testing <HomeScreen />', () => {

    const user = {
        name: 'Aaron',
        email: 'aaron@gmail.com'
    }

    // * Testing useContext -> render w/ mount() wrapping child component with UserContext.Provider/HigherOrderComponent
    const wrapper = mount(
        <UserContext.Provider value={ {user} }>
            <HomeScreen />
        </UserContext.Provider>
    );

    test('should match snapshot', () => {
        expect( wrapper ).toMatchSnapshot();

    });

});