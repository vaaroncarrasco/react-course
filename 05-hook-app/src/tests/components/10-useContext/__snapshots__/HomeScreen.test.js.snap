// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <HomeScreen /> should match snapshot 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Aaron",
   "email": "aaron@gmail.com"
}
  </pre>
</div>
`;
